dataprofiler/profilers/profiler_options.py

```diff
@@ -895,6 +905,34 @@ def _validate_helper(self, variable_path: str = "CategoricalOptions") -> list[st
                 "{}.top_k_categories must be either None"
                 " or a positive integer".format(variable_path)
             )
+
+        if self.max_sample_size_to_check_stop_condition is not None and (
+            not isinstance(self.max_sample_size_to_check_stop_condition, int)
+            or self.max_sample_size_to_check_stop_condition < 0
+        ):
+            errors.append(
+                "{}.max_sample_size_to_check_stop_condition must be either None"
+                " or a non-negative integer".format(variable_path)
+            )
+
+        if self.stop_condition_unique_value_ratio is not None and (
+            not isinstance(self.stop_condition_unique_value_ratio, float)
+            or self.stop_condition_unique_value_ratio < 0
+        ):
```

Comment:
Because it is a ratio we need to: `0 <= stop_condition_unique_value_ratio <= 1.0`

Reply:
ah I knew it didn't makes sense for a hard core `< 0`